shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/DALBackendHandlerFactoryTest.java

```diff
@@ -0,0 +1,104 @@
+        assertFieldOfInstance(useDatabaseBackendHandler, "useStatement", is(mySQLUseStatement));
+        assertFieldOfInstance(useDatabaseBackendHandler, "backendConnection", is(backendConnection));
+    }
+    
+    @Test
+    public void assertShowDatabasesBackendHandlerReturnedWhenMySQLShowDatabasesStatement() {
+        BackendConnection backendConnection = mock(BackendConnection.class);
+        TextProtocolBackendHandler textProtocolBackendHandler = DALBackendHandlerFactory.newInstance("", mock(MySQLShowDatabasesStatement.class), backendConnection);
+        assertThat(textProtocolBackendHandler, instanceOf(ShowDatabasesBackendHandler.class));
+        ShowDatabasesBackendHandler showDatabasesBackendHandler = (ShowDatabasesBackendHandler) textProtocolBackendHandler;
+        assertFieldOfInstance(showDatabasesBackendHandler, "backendConnection", is(backendConnection));
+    }
+    
+    @Test
+    public void assertBroadcastBackendHandlerReturnedWhenSetStatement() {
+        SetStatement setStatement = mock(SetStatement.class);
+        BackendConnection backendConnection = mock(BackendConnection.class);
+        TextProtocolBackendHandler textProtocolBackendHandler = DALBackendHandlerFactory.newInstance("", setStatement, backendConnection);
+        assertThat(textProtocolBackendHandler, instanceOf(BroadcastBackendHandler.class));
+        BroadcastBackendHandler broadcastBackendHandler = (BroadcastBackendHandler) textProtocolBackendHandler;
+        assertFieldOfInstance(broadcastBackendHandler, "sqlStatement", is(setStatement));
+        assertFieldOfInstance(broadcastBackendHandler, "sql", is(""));
+        assertFieldOfInstance(broadcastBackendHandler, "backendConnection", is(backendConnection));
+    }
+    
+    @Test
+    public void assertUnicastBackendHandlerReturnedWhenOtherDALStatement() {
+        DALStatement dalStatement = mock(DALStatement.class);
+        BackendConnection backendConnection = mock(BackendConnection.class);
+        TextProtocolBackendHandler textProtocolBackendHandler = DALBackendHandlerFactory.newInstance("", dalStatement, backendConnection);
+        assertThat(textProtocolBackendHandler, instanceOf(UnicastBackendHandler.class));
+        UnicastBackendHandler unicastBackendHandler = (UnicastBackendHandler) textProtocolBackendHandler;
+        assertFieldOfInstance(unicastBackendHandler, "sqlStatement", is(dalStatement));
+        assertFieldOfInstance(unicastBackendHandler, "sql", is(""));
+        assertFieldOfInstance(unicastBackendHandler, "backendConnection", is(backendConnection));
+    }
+    
+    @SneakyThrows
+    private <S, T> void assertFieldOfInstance(final S classInstance, final String fieldName, final Matcher<T> matcher) {
+        Field field = classInstance.getClass().getDeclaredField(fieldName);
+        field.setAccessible(true);
+        T value = (T) field.get(classInstance);
+        MatcherAssert.assertThat(value, matcher);
```

Comment:
Please use static import of MatcherAssert

Reply:
changed